Mathlib/Data/Seq/Seq.lean

```diff
@@ -489,7 +545,7 @@ def drop (s : Seq α) : ℕ → Seq α
   | 0 => s
   | n + 1 => tail (drop s n)
 
-attribute [simp] drop
+-- attribute [simp] drop
```

Comment:
What's the reasoning here?

Reply:
When `drop` is marked with `[simp]`, the simplifier unfolds `(cons x s).drop (n + 1)` as `((cons x s).drop n).tail` instead of `s.drop n`, even when the `drop_succ_cons` simp-lemma is present. In general, I think that replacing `s.drop (n + 1)` with `(s.drop n).tail` is not what the user wants by default because it adds another sequence operation to the expression, which rarely is simplification.

With `drop_nil` and `drop_succ_cons` instead, the simplifier always simplifies the expression.